v3/src/models/data/data-set-undo.ts

```diff
@@ -32,3 +32,25 @@ export const setCaseValuesCustomUndoRedo: ICustomUndoRedoPatcher = {
     }
   }
 }
+
+export function setCaseValuesWithCustomUndoRedo(data: IDataSet, cases: ICase[], affectedAttributes?: string[]) {
+  const items: IItem[] = []
+  cases.forEach(aCase => {
+    // convert each parent case change to a change to each underlying item
+    const caseGroup = data.caseGroupMap.get(aCase.__id__)
+    if (caseGroup?.childCaseIds?.length) {
+      items.push(...caseGroup.childItemIds.map(id => ({ ...aCase, __id__: id })))
+    }
```

Comment:
```suggestion
    if (caseGroup?.childCaseIds?.length) {
      items.push(...caseGroup.childItemIds.map(id => ({ ...aCase, __id__: id })))
    } else {
      items.push(aCase)
    }
```
This would allow you to update both cases and items in the same call to `setCaseValuesWithCustomUndoRedo`. But if there's some reason to not do this, I think it would be worth including a comment that says that if you include any cases, items specified will not be updated.